internal/elasticsearch/ingest/datastream.go

```diff
@@ -75,12 +92,43 @@ func loadIngestPipelineFiles(dataStreamPath string, nonce int64) ([]Pipeline, er
 			Path:    path,
 			Name:    getPipelineNameWithNonce(name[:strings.Index(name, ".")], nonce),
 			Format:  filepath.Ext(path)[1:],
-			Content: c,
+			Content: append(c, routingPipeline...),
 		})
 	}
 	return pipelines, nil
 }
 
+func loadRoutingRuleFile(dataStreamPath string) ([]byte, error) {
+	routingRulePath := filepath.Join(dataStreamPath, "routing_rules.yml")
+	c, err := os.ReadFile(routingRulePath)
+	if err != nil {
+		return nil, fmt.Errorf("reading routing_rules.yml failed (path: %s): %w", routingRulePath, err)
+	}
+
+	// unmarshal yaml into a struct
+	var data []SourceData
+	err = yaml.Unmarshal(c, &data)
+	if err != nil {
+		return nil, fmt.Errorf("unmarshalling routing_rules.yml content failed: %w", err)
+	}
+
+	// Now you can work with the data as Go structs
+	constructPipeline := ""
+	for _, srcData := range data {
+		for _, rule := range srcData.Rules {
+			rerouteProcessor := "  - reroute:\n"
+			rerouteProcessor += "      tag: " + srcData.SourceDataset.(string) + "\n"
+			rerouteProcessor += "      if: " + rule.If.(string) + "\n"
+			//TODO: deal with multiple datasets?
+			rerouteProcessor += "      dataset: " + rule.TargetDataset.(string) + "\n"
+			//TODO: deal with multiple namespaces?
+			rerouteProcessor += "      namespace: " + rule.Namespace[0].(string) + "\n"
```

Comment:
We must support multiple datasets and namespaces.

Reply:
yep added!